execute_test.go

```diff
@@ -233,6 +234,411 @@ func TestExecutor_emptyPlansWithDependencies(t *testing.T) {
 	}, result)
 }
 
+func TestExecutor_insertIntoFragment(t *testing.T) {
+	// the query we want to execute is
+	// {
+	//   photo {								<- Query.services @ serviceA
+	//     ... photoFragment
+	//     }
+	//   }
+	// }
+	// fragment createdByFragment on Photo {
+	//   createdBy {								<- User boundary type
+	// 	 	firstName								<- User.firstName @ serviceA
+	//   	address 							<- User.address @ serviceB
+	//   }
+	// }
+	result, err := (&ParallelExecutor{}).Execute(&ExecutionContext{
```

Comment:
I was able to come up with these tests by following this formula:

1. The plan should only include `selectionSets` for fields that are being selected in the query. I.e. do not include `selectionSets` from auxiliary queries like the ones for the `node` query needed for federation
2. For fragments spreads, the `selectionSet` at a particular step should include the name of the `fragmentSpread`, while the actual selection of fields should be in the `FragmentDefinitionsList` of the step (according to step #1 above)
3. For federated queries, the expected returned value of the *parent node that contains a boundary field* should be of the form:
```
{
    "boundaryType": {
        < any fields selected that are not federated. I.e. fields that are in the same service as "boundaryType" >
        "id": "<some-id>",
    }
}
```
4. For federated queries, the expected returned value of the *node that resolves a boundary field* should be of the form:
```
{
    "node": {
         "<boundaryField>": "<boundaryValue>"
     }
}
```
5. For federated queries, the expected value received in the `Queryer` of the *node that resolves a boundary field* should include the `"id"` returned in the parent node (in this case `"<some-id>"`)

Reply:
This sounds exactly right. Nice job figuring this out without a lot of docs!!